### PR TITLE
Add CoNLL-2003 shared task dataset

### PR DESCRIPTION
**Pull request by vblagoje**
Please consider adding CoNLL-2003 shared task dataset as it's beneficial for token classification tasks. The motivation behind this PR is the [PR](https://github.com/huggingface/transformers/pull/7041) in the transformers project. This dataset would be not only useful for the usual run-of-the-mill NER tasks but also for syntactic chunking and part-of-speech (POS) tagging.

### CODE REVIEW

File: datasets/conll2003/conll2003.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "word": datasets.Sequence(datasets.Value("string")),
+                    "pos": datasets.Sequence(datasets.Value("string")),
+                    "chunk": datasets.Sequence(datasets.Value("string")),
+                    "ner": datasets.Sequence(datasets.Value("string")),
+                }
```
Comment (lhoestq) [inline datasets/conll2003/conll2003.py:86]: We have other POS/NER datasets (xtreme/pan-x and Lince)and they name their features "tokens"/"words" and "pos" and "ner"/"ner_tags". I would be cool to have the same names everywhere.

What about using the convention "words" + "pos" + "ner" ?
For CoNNL-2003 there would only be one change: word -> words

Then we could update Lince and xtreme/pan-x

WDYT @thomwolf ?

**Review by lhoestq — COMMENTED** [review]
Thank you for adding this one :) 

Everything is fine to me.
Before merging I'd like to discuss a bit the naming convention for the features:

File: datasets/conll2003/conll2003.py (hunk 1 above)
Comment (thomwolf) [reply datasets/conll2003/conll2003.py:86]: I'm fine with that

**Comment by stefan-it** [comment]
I think we should somewhere mention, that is the dataset in IOB2 tagging scheme, whereas the original dataset uses IOB1 :)

File: datasets/conll2003/conll2003.py (hunk 1 above)
Comment (lhoestq) [reply datasets/conll2003/conll2003.py:86]: Ok I'm opening a PR for that

File: datasets/conll2003/conll2003.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+The shared task of CoNLL-2003 concerns language-independent named entity recognition. We will concentrate on
+four types of named entities: persons, locations, organizations and names of miscellaneous entities that do
+not belong to the previous three groups.
+
+The CoNLL-2003 shared task data files contain four columns separated by a single space. Each word has been put on
+a separate line and there is an empty line after each sentence. The first item on each line is a word, the second
+a part-of-speech (POS) tag, the third a syntactic chunk tag and the fourth the named entity tag. The chunk tags
+and the named entity tags have the format I-TYPE which means that the word is inside a phrase of type TYPE. Only
+if two phrases of the same type immediately follow each other, the first word of the second phrase will have tag
+B-TYPE to show that it starts a new phrase. A word with tag O is not part of a phrase.
```
Comment (lhoestq) [inline datasets/conll2003/conll2003.py:46]:
```suggestion
a part-of-speech (POS) tag, the third a syntactic chunk tag and the fourth the named entity tag (IOB2). The chunk
tags and the named entity tags have the format I-TYPE which means that the word is inside a phrase of type TYPE.
Only if two phrases of the same type immediately follow each other, the first word of the second phrase will have
tag B-TYPE to show that it starts a new phrase. A word with tag O is not part of a phrase.
```

File: datasets/conll2003/dataset_infos.json [hunk 3]
```diff
@@ -0,0 +1 @@
..._processed": null, "supervised_keys": null, "builder_name": "conll2003", "config_name": "conll2003", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 6032006, "num_examples": 14041, "dataset_name": "conll2003"}, "validation": {"name": "validation", "num_bytes": 1512711, "num_examples": 3250, "dataset_name": "conll2003"}, "test": {"name": "test", "num_bytes": 1378578, "num_examples": 3453, "dataset_name": "conll2003"}}, "download_checksums": {"https://github.com/davidsbatista/NER-datasets/raw/master/CONLL2003/train.txt": {"num_bytes": 3283418, "checksum": "c99b26852dabf57ca9d30a0e892b84544cc8962003151e14a71077c55dc66db5"}, "https://github.com/davidsbatista/NER-datasets/raw/master/CONLL2003/valid.txt": {"num_bytes": 827441, "checksum": "f1f6469322876887de1d04acd43c59b02f59d5b02acf42c027132fa1bf349cb2"}, "https://github.com/davidsbatista/NER-datasets/raw/master/CONLL2003/test.txt": {"num_bytes": 748093, "checksum": "82e0c72d262f86ad3e78b15c5d980bbf87cb205aa4bf6d2d97643f463f8d7ff7"}}, "download_size": 4858952, "post_processing_size": null, "dataset_size": 8923295, "size_in_bytes": 13782247}}
```
Comment (lhoestq) [inline datasets/conll2003/dataset_infos.json:1]:
```suggestion
{"conll2003": {"description": "The shared task of CoNLL-2003 concerns language-independent named entity recognition. We will concentrate on \nfour types of named entities: persons, locations, organizations and names of miscellaneous entities that do \nnot belong to the previous three groups. \n\nThe CoNLL-2003 shared task data files contain four columns separated by a single space. Each word has been put on \na separate line and there is an empty line after each sentence. The first item on each line is a word, the second \na part-of-speech (POS) tag, the third a syntactic chunk tag and the fourth the named entity tag (IOB2). The chunk \ntags and the named entity tags have the format I-TYPE which means that the word is inside a phrase of type TYPE. \nOnly if two phrases of the same type immediately follow each other, the first word of the second phrase will have \ntag B-TYPE to show that it starts a new phrase. A word with tag O is not part of a phrase.\n\nFor more details see https://www.clips.uantwerpen.be/conll2003/ner/ and https://www.aclweb.org/anthology/W03-0419\n", "citation": "@inproceedings{tjong-kim-sang-de-meulder-2003-introduction,\n    title = \"Introduction to the {C}o{NLL}-2003 Shared Task: Language-Independent Named Entity Recognition\",\n    author = \"Tjong Kim Sang, Erik F.  and\n      De Meulder, Fien\",\n    booktitle = \"Proceedings of the Seventh Conference on Natural Language Learning at {HLT}-{NAACL} 2003\",\n    year = \"2003\",\n    url = \"https://www.aclweb.org/anthology/W03-0419\",\n    pages = \"142--147\",\n}\n", "homepage": "https://www.aclweb.org/anthology/W03-0419/", "license": "", "features": {"id": {"dtype": "string", "id": null, "_type": "Value"}, "word": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "pos": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "chunk": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}, "ner": {"feature": {"dtype": "string", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}}, "post_processed": null, "supervised_keys": null, "builder_name": "conll2003", "config_name": "conll2003", "version": {"version_str": "1.0.0", "description": null, "major": 1, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 6032006, "num_examples": 14041, "dataset_name": "conll2003"}, "validation": {"name": "validation", "num_bytes": 1512711, "num_examples": 3250, "dataset_name": "conll2003"}, "test": {"name": "test", "num_bytes": 1378578, "num_examples": 3453, "dataset_name": "conll2003"}}, "download_checksums": {"https://github.com/davidsbatista/NER-datasets/raw/master/CONLL2003/train.txt": {"num_bytes": 3283418, "checksum": "c99b26852dabf57ca9d30a0e892b84544cc8962003151e14a71077c55dc66db5"}, "https://github.com/davidsbatista/NER-datasets/raw/master/CONLL2003/valid.txt": {"num_bytes": 827441, "checksum": "f1f6469322876887de1d04acd43c59b02f59d5b02acf42c027132fa1bf349cb2"}, "https://github.com/davidsbatista/NER-datasets/raw/master/CONLL2003/test.txt": {"num_bytes": 748093, "checksum": "82e0c72d262f86ad3e78b15c5d980bbf87cb205aa4bf6d2d97643f463f8d7ff7"}}, "download_size": 4858952, "post_processing_size": null, "dataset_size": 8923295, "size_in_bytes": 13782247}}
```
updated the dataset_info as well

**Comment by lhoestq** [comment]
Indeed this is something we want to mention.

If would want to add more details about the IOB1->2 change, feel free to ignore my suggestions and edit the description + update the dataset_info

**Comment by vblagoje** [comment]
@lhoestq do you want me to update it or you'll update it. I am ok either way

**Comment by lhoestq** [comment]
The best would be to mention this change in the description and then update the dataset_info.json file.
Could you do that if you don't mind ?

Then it should be ready to merge :)

Thanks again for adding this dataset !

**Comment by vblagoje** [comment]
No problem @lhoestq I'll do the update

**Comment by vblagoje** [comment]
@lhoestq please check if 847addf is exactly what we want

**Review by lhoestq — APPROVED** [review]
Perfect, thank you !

**Comment by Nilabhra** [comment]
Is the German task also part of this? If not, can it be accessed via the Datasets library?

**Pull request merged**